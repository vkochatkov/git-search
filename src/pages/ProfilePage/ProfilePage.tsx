import React, { useEffect } from 'react';
import { getUser, getRepos } from '../../store/actionCreators/actionCreators';
import { useDispatch, useSelector } from 'react-redux';
import { Link, useParams } from 'react-router-dom';
import { Props, UserState, ReposProps } from '../../interfaces/interface';
import { Repos } from '../../components/Repos/Repos';

interface ParamsName {
  urlName: string;
}

export const ProfilePage: React.FC<Props> = () => {
  const user = useSelector((state: UserState) => state.user);
  const repos = useSelector((state: ReposProps) => state.repos);

  const dispatch = useDispatch();
  const { urlName } = useParams<ParamsName>();

  const {
    name,
    avatar_url,
    company,
    location,
    bio,
    blog,
    login,
    html_url,
    followers,
    public_repos,
    public_gists,
    following,
  } = user;

  useEffect(() => {
    dispatch(getRepos(urlName));
    dispatch(getUser(urlName));
  }, []);

  return (
    <>
      <Link to={'/'} className={'btn btn-link'}>
        На главную
      </Link>
      <div className={'card mb-4'}>
        <div className="card-body">
          <div className="row">
            <div className="col-sm-3 text-center">
              <img src={avatar_url} alt={name} style={{ width: '150px' }} />

              <h1>{name}</h1>
              {location && <p>Местоположение: {location}</p>}
            </div>
            <div className="col">
              {bio && (
                <>
                  <h3>BIO</h3>
                  <p>{bio}</p>
                </>
              )}
              <a
                href={html_url}
                className={'btn btn-dark'}
                rel="noreferrer"
                target={'_blank'}
              >
                Открыть профиль
              </a>
              <ul>
                {login && (
                  <li>
                    <strong>Username: </strong> {login}
                  </li>
                )}
                {company && (
                  <li>
                    <strong>Company: </strong> {company}
                  </li>
                )}
                {blog && (
                  <li>
                    <strong>Website: </strong> {blog}
                  </li>
                )}
              </ul>
              <div className={'badge badge-primary'}>
                Подписчики: {followers}
              </div>
              <div className={'badge badge-success'}>Подписан: {following}</div>
              <div className={'badge badge-info'}>
                Репозитории: {public_repos}
              </div>
              <div className={'badge badge-dark'}>Gists: {public_gists}</div>
            </div>
          </div>
        </div>
      </div>
      <Repos repos={repos} />
    </>
  );
};
