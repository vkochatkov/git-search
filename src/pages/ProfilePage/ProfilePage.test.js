import React from 'react';
import { configure } from 'enzyme';
import { render } from '@testing-library/react';
import { ProfilePage } from './ProfilePage';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import { BrowserRouter, Link } from 'react-router-dom';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';

configure({ adapter: new Adapter() });

const setUp = ({ store }) => {
  return render(
    <Provider store={store}>
      <BrowserRouter>
        <ProfilePage />
      </BrowserRouter>
    </Provider>,
  );
};

describe('ProfilePage component', () => {
  const initialState = {
    user: { id: 1 },
    repos: [
      {
        id: 2,
        html_url: 'https://avatars2.githubusercontent.com/u/39588133?v=4',
        name: '"fafaf"',
      },
    ],
  };
  const middlewares = [thunk];
  const mockStore = configureStore(middlewares);
  let store;

  it('should render ProfilePage', () => {
    store = mockStore(initialState);
    const wrapper = setUp({ store });
    expect(wrapper).toMatchSnapshot();
  });
});
