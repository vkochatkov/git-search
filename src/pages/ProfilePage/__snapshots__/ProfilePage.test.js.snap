// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfilePage component should render ProfilePage 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="btn btn-link"
        href="/"
      >
        На главную
      </a>
      <div
        class="card mb-4"
      >
        <div
          class="card-body"
        >
          <div
            class="row"
          >
            <div
              class="col-sm-3 text-center"
            >
              <img
                style="width: 150px;"
              />
              <h1 />
            </div>
            <div
              class="col"
            >
              <a
                class="btn btn-dark"
                rel="noreferrer"
                target="_blank"
              >
                Открыть профиль
              </a>
              <ul />
              <div
                class="badge badge-primary"
              >
                Подписчики: 
              </div>
              <div
                class="badge badge-success"
              >
                Подписан: 
              </div>
              <div
                class="badge badge-info"
              >
                Репозитории: 
              </div>
              <div
                class="badge badge-dark"
              >
                Gists: 
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="card mb-3"
      >
        <div
          class="card-body"
        >
          <h5>
            <a
              href="https://avatars2.githubusercontent.com/u/39588133?v=4"
              rel="noreferrer"
              target="_blank"
            >
              "fafaf"
            </a>
          </h5>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <a
      class="btn btn-link"
      href="/"
    >
      На главную
    </a>
    <div
      class="card mb-4"
    >
      <div
        class="card-body"
      >
        <div
          class="row"
        >
          <div
            class="col-sm-3 text-center"
          >
            <img
              style="width: 150px;"
            />
            <h1 />
          </div>
          <div
            class="col"
          >
            <a
              class="btn btn-dark"
              rel="noreferrer"
              target="_blank"
            >
              Открыть профиль
            </a>
            <ul />
            <div
              class="badge badge-primary"
            >
              Подписчики: 
            </div>
            <div
              class="badge badge-success"
            >
              Подписан: 
            </div>
            <div
              class="badge badge-info"
            >
              Репозитории: 
            </div>
            <div
              class="badge badge-dark"
            >
              Gists: 
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="card mb-3"
    >
      <div
        class="card-body"
      >
        <h5>
          <a
            href="https://avatars2.githubusercontent.com/u/39588133?v=4"
            rel="noreferrer"
            target="_blank"
          >
            "fafaf"
          </a>
        </h5>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
